### PR TITLE
Fix double-quoted warnings in Xcode 12

### PR DESCRIPTION
**Pull request by AnastasiaKubova**
Things to consider before you submit the PR:

* [x] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [ ] ~Did you add unit tests?~
* [x] Did you test your change with the sample apps?

## Description

Enable this warning in the scripts and fix headers.

## Related PRs or issues

[AB#83626](https://msmobilecenter.visualstudio.com/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/83626)

### CODE REVIEW

File: Source/CrashReporter.h [hunk 1]
```diff
@@ -34,12 +34,30 @@
 
 // This must be included before any other PLCrashReporter includes, as
 // it redefines symbol names
+#if __has_include(<CrashReporter/PLCrashNamespace.h>)
```
Comment (MatkovIvan) [inline Source/CrashReporter.h:37]: no need for multiple `#if` here. You can check it just once

File: Source/PLCrashReportThreadInfo.m [hunk 2]
```diff
@@ -26,7 +26,10 @@
  * OTHER DEALINGS IN THE SOFTWARE.
  */
 
+#ifndef PLCRASH_REPORT_THREAD_INFO
+#define PLCRASH_REPORT_THREAD_INFO
 #import "PLCrashReportThreadInfo.h"
+#endif
```
Comment (MatkovIvan) [inline Source/PLCrashReportThreadInfo.m:32]: Wrong guard usage. It should be only inside headers. Not in placed where you import it

File: Source/PLCrashReport.h [hunk 3]
```diff
@@ -28,19 +28,83 @@
 
 #import <Foundation/Foundation.h>
 
+#if __has_include(<CrashReporter/PLCrashReportApplicationInfo.h>)
```
Comment (MatkovIvan) [inline Source/PLCrashReport.h:31]: if this file is umbrella for framework - no need for checking at all. If this used in static libraries - we can make separate file for it

**Review by DmitriyKirakosyan — DISMISSED** [review]
Tested with Pods, SPM, and adding it manually. All work as expected without warnings.

File: Source/CrashReporter.h [hunk 4]
```diff
@@ -32,14 +32,25 @@
 #import <AvailabilityMacros.h>
 #endif
 
+#if __has_include(<CrashReporter/PLCrashNamespace.h>) \
+    && __has_include(<CrashReporter/PLCrashReporter.h>) \
+    && __has_include(<CrashReporter/PLCrashReport.h>) \
+    && __has_include(<CrashReporter/PLCrashReportTextFormatter.h>)
```
Comment (MatkovIvan) [inline Source/CrashReporter.h:38]:
```suggestion
#if __has_include(<CrashReporter/PLCrashReport.h>)
```

File: Source/PLCrashReport.h [hunk 5]
```diff
@@ -28,6 +28,36 @@
 
 #import <Foundation/Foundation.h>
 
+#ifndef PLCRASH_REPORT
+#define PLCRASH_REPORT
+
+#if __has_include(<CrashReporter/PLCrashReportApplicationInfo.h>) \
```
Comment (MatkovIvan) [inline Source/PLCrashReport.h:34]: Simplify these conditions

File: Source/PLCrashReport.h [hunk 6]
```diff
@@ -28,6 +28,36 @@
 
 #import <Foundation/Foundation.h>
 
+#ifndef PLCRASH_REPORT
```
Comment (MatkovIvan) [inline Source/PLCrashReport.h:31]:
```suggestion
#ifndef PLCRASH_REPORT_H
```

File: Source/PLCrashReport.pb-c.h [hunk 7]
```diff
@@ -10,7 +10,7 @@ PROTOBUF_C__BEGIN_DECLS
 
 #if PROTOBUF_C_VERSION_NUMBER < 1000000
 # error This file was generated by a newer version of protoc-c which is incompatible with your libprotobuf-c headers. Please update your headers.
-#elif 1003003 < PROTOBUF_C_MIN_COMPILER_VERSION
+#elif 1003002 < PROTOBUF_C_MIN_COMPILER_VERSION
```
Comment (MatkovIvan) [inline Source/PLCrashReport.pb-c.h:13]: Not related, revert

**Review by MatkovIvan — CHANGES_REQUESTED** [review]

File: Source/PLCrashReporterNSError.h [hunk 8]
```diff
@@ -27,9 +27,16 @@
  */
 
 #import <Foundation/Foundation.h>
-#import "CrashReporter.h"
 
+#if __has_include(<CrashReporter/CrashReporter.h>) \
+    && __has_include(<CrashReporter/PLCrashMacros.h>)
```
Comment (MatkovIvan) [inline Source/PLCrashReporterNSError.h:32]:
```suggestion
#if __has_include(<CrashReporter/CrashReporter.h>)
```

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,5 +1,9 @@
 # PLCrashReporter Change Log
 
+## Version 1.8.2 (Under development)
+
+* **[Fix]** Fix `double-quoted` warnings in Xcode 12.
```
Comment (Jamminroot) [inline CHANGELOG.md:5]:
```suggestion
* **[Fix]** Fix `double-quoted imports` warnings in Xcode 12.
```

Comment (AnastasiaKubova) [reply CHANGELOG.md:5]: The full warning looks `Double-quoted include "SameHeader.h" in framework header, expected angle-bracketed instead.`, so I am not sure that use `imports` will be correct in this case. Besides, we already have the same changelog [here]( https://github.com/microsoft/appcenter-sdk-apple/blob/develop/CHANGELOG.md#version-401-under-development). Let's keep it as is.

**Review by MatkovIvan — APPROVED** [review]

**Review by DmitriyKirakosyan — APPROVED** [review]

**Pull request merged**